### DISCUSSION TITLE
Docs: Fabric

### DISCUSSION THREAD

**Discussion by cortinico** [reactwg/react-native-new-architecture#9] [Documentation]
Hi all,
This thread is to **collect feedback** on the section of the website called: Fabric - https://reactnative.dev/docs/next/fabric-renderer

If you happen to go through the guide, feel free to report:
* Any part that is harder to follow.
* Any part that is incorrect or not described as you would expect.
* Any improvements or missing piece.

General feedback is also more than welcome.

**Comment by prabakarviji** [comment]
It looks good except  **Faster Startup**, will be more helpful if we provide some more info about it.

**Reply by cortinico** [reply]
Could you elaborate on what is not clear or what you would like to see more of? Essentially _Faster Startup_ was a synonym to express the lazy initialization.

**Reply by prabakarviji** [reply]
Yup. That's the thing. A hint about lazy initialization will be more helpful 🙏

**Reply by cortinico** [reply]
It's written on the same line: "Faster Startup: Host components are lazily initialized by default.".
Were you referring to something different?

**Reply by prabakarviji** [reply]
Got it :)

**Comment by p-syche** [comment]
This section of the docs reads really well, even though it's pretty heavy in the amount of knowledge it passes.
As a general feedback I would love to see little popup boxes when hovering over items from the Glossary. Like on Wikipedia 😄  

The problem is that when I'm reading something and I see a link, I don't want to loose track of what I'm reading so I open the link in a new tab. After reading the main text I check out the other tabs and I'm often wondering: "why did I open this?" XD
Having popup boxes with the first sentence or 2 from the glossary would make this experience smoother 🍸

**Reply by cortinico** [reply]
That would be a nice feature and I believe would improve the reader experience. @Simek do you happen to know if this is doable with Docusaurus in any form?

**Reply by Simek** [reply]
I have no knowledge of such feature or plugin for the Docusaurus, but I will research the possibilities.

**Comment by chakrihacker** [comment]
> The new architecture also provides benefits in code quality

Can we provide an example comparing old renderer and new renderer on how it improves code quality?

If we consider type safety as code quality improvement can we rephrase to explain it?

**Reply by cortinico** [reply]
> If we consider type safety as code quality improvement can we rephrase to explain it?

Yes that's what we consider as "code quality improvement". It's explained as the first point in the bullet list below. Isn't that clear enough?

**Reply by sammy-SC** [reply]
Hey @chakrihacker,

that's a great question! I agree it lacks specificity and should include examples. Code quality is hard to measure and comes down to individual preferences. I've noted this down and will think about it.

Type safety is one example of improved code quality. 
Another example would be synchronous nature of Fabric and how some operations no longer requires thread switching. This makes code easier to reason about and less complex. Checkout ["measure" in Fabric](https://github.com/facebook/react-native/blob/main/ReactCommon/react/renderer/uimanager/UIManager.cpp#L253-L287) where Fabric changes thread of execution zero times. Compare it to ["measure" in the old architecture](https://github.com/facebook/react-native/blob/main/React/Modules/RCTUIManager.m#L1305-L1336). The old architecture changes thread of execution two times. By measure, I'm referring to a set of methods dedicated to retrieving layout metrics.

On iOS, we took the opportunity and rewrote all components from scratch for Fabric. We would like to think the components are written in cleaner and more maintainable manner but this depends on individual's taste.
